css-view-transitions-1/Overview.bs

```diff
@@ -197,38 +197,66 @@ urlPrefix: https://html.spec.whatwg.org/multipage/rendering.html; type: dfn;
 
 # Pseudo-elements # {#pseudo}
 
-	The following <dfn export>view-transition pseudo-elements</dfn> represent the various items being animated.
+## Pseudo-element parents ## {#parent-pseudo}
 
-	The ''::view-transition'' pseudo-element acts as a grouping element for other [=view-transition pseudo-elements=]
-	and has the document's [=document element=] as its [=originating element=].
+	Issue: This definition should be moved to [[css-pseudo-4]].
 
-	Note: For example, '':root::view-transition'' selector matches this pseudo-element,
-	but ''div::view-transition'' does not.
+	A <dfn>pseudo-element parent</dfn> is a type of [=tree-abiding pseudo-element=].
 
-	Other [=view-transition pseudo-elements=] take a <<pt-name-selector>> argument
-	to specify which elements named with 'view-transition-name' are affected.
+	It has <dfn for="pseudo-element parent">children</dfn>,
+	a [=/list=] of [=pseudo-elements=].
+	Initially an empty [=/list=].
 
-	There can be multiple pseudo-elements of the same type,
-	one for each 'view-transition-name' participating in a transition.
+	A [=pseudo-element parent=]'s [=pseudo-element parent/children=] defines the nested children of the [=pseudo-element parent=] in document order.
 
-	The <<pt-name-selector>> is defined as follows:
+	When a [=pseudo-element=] |pseudo| is part of a [=pseudo-element parent=] |parent|'s [=pseudo-element parent/children=],
+	then |pseudo|'s [=originating pseudo-element=] is |parent|.
+
+	If a [=pseudo-element parent=]'s [=pseudo-element parent/children=]'s [=list/size=] is 1, then '':only-child'' selects [=pseudo-element parent/children=][0].
```

Comment:
How should only-child work for ::view-transitions whose originating element is html. If we consider ::view-transition to be a child, similar to other DOM children, then should it always match? I can't think of a case where there are no nodes under html. Even if its possible, it'll be simpler if we don't have to support that.

Reply:
The more I think about it, I think `::view-transition:only-child` should be 'invalid'.

`element:only-child` ignores pseudo-elements, so I don't think there's a sensible behaviour for `::pseudo:only-child` when the pseudo is a child of an element.

Reply:
> element:only-child ignores pseudo-elements, so I don't think there's a sensible behaviour for ::pseudo:only-child when the pseudo is a child of an element.

That makes sense. And lines up with @vmpstr's suggestion + the implementation.